### ISSUE TITLE
Unable to Retrieve 'data' Property from PopupMenuItem in Version 0.21.1

### ISSUE THREAD

**Issue by Klu1d** [flet-dev/flet#2826]
**Issue:** 
When I updated flet to version 0.21.1, I encountered the issue that it became impossible to retrieve the 'data' property when clicking on PopupMenuItem. Prior to this, in version 0.20.0, I could retrieve this property from PopupMenuItem.

**Demonstrations:**
[flet_version_0.20.0.webm](https://github.com/flet-dev/flet/assets/105982046/f07d7c9d-6034-44ad-a601-6bb596d14d9e)
[flet_version_0.21.1.webm](https://github.com/flet-dev/flet/assets/105982046/242b5e27-1804-42e4-b1c7-ad2a74109c07)

**Code:**
```
import flet as ft

def main(page: ft.Page):
    def lola(e):
        print('Current flet Version 0.21.1: ', e.control.data)
    options = ft.PopupMenuButton(
            icon=ft.icons.MORE_HORIZ_OUTLINED,
            visible=True,
            items=[
                ft.PopupMenuItem(text="change", data='tata'),
                ft.PopupMenuItem(text="delete", data='tata', on_click=lola),
            ]
        )
    
    page.add(options)

ft.app(target=main)
```

**Comment by FeodorFitsner** [comment]
Sorry, what was the issue?

**Comment by Klu1d** [comment]
I apologize for not providing a response earlier. I'm not entirely sure what the issue was, but I made some changes to the project architecture and reinstalled the latest version of Flet. I've come to the conclusion that the problem wasn't with Flet itself, but rather with how I wrote the code. 

Despite checking the code with different versions and isolating the provided code from my project, the problem was still resolved by reinstalling: pip install -U flet.
